### PR TITLE
Adding Functions

### PR DESCRIPTION
**Pull request by lerfast**
Hi, I did the following tasks:

- [ ] Start my project for the second milestone.
- [ ] You can add the task to start the list.
- [ ] You can remove all of the completed tasks

image

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -50,4 +30,95 @@ const renderList = (list) => {
...
+
+const editTaskDescription = (taskId, newDescription) => {
+  todoList.forEach((task) => {
+    if (task.id === taskId) {
+      task.description = newDescription;
+    }
+  });
+  saveToLocalStorage(todoList);
+  renderList(todoList);
+};
```
Comment (DuaneDave) [inline src/index.js]: I can see that you have implemented the edit functionality here, however, it is not working as expected, consider making the item editable when it is clicked upon in order to make it possible for a user to perform edits

File: src/index.js [hunk 2]
```diff
@@ -8,38 +8,18 @@ const reloadIcon = document.getElementById('reload-img');
...
   } else {
-    const sortedList = list.sort((a, b) => a.id - b.id);
+    const sortedList = list.sort((a, b) => a.index - b.index);
     sortedList.forEach((task) => {
       innerList += `
-          <li class="to-do-tasks">
+          <li class="to-do-tasks" data-task-id="${task.id}">
             <div class="check-box ${task.completed ? 'completed' : ''}">${task.completed ? '✓' : ''}</div>
             <p class="task-description ${task.completed ? 'line-through' : ''}">${task.description}</p>
             <img class="more-logo" src="${moreIcon}"/>
```
Comment (DuaneDave) [inline src/index.js]: This icon, when clicked should at least show an icon that makes deleting a single todo possible, I recommend including this as it is a requirement

File: src/index.js (hunk 1 above)
Comment (DuaneDave) [inline src/index.js]: I can see that you have implemented the edit functionality here, however, it is not working as expected, consider making the item editable when it is clicked upon in order to make it possible for a user to perform edits

**Review by DuaneDave — CHANGES_REQUESTED** [review]
Hi @lerfast 👋 , 

Good job so far on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ The clear completed button working fine
✔️ Items are saved to local storage as expected

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (lerfast) [reply src/index.js]: Fix it, thanks!!!

image

File: src/index.js (hunk 2 above)
Comment (lerfast) [reply src/index.js]: Dit it!!!, thanks!!!

image

**Review by codecaiine — APPROVED** [review]
Hi @lerfast , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Editing function is well implemented and works as expected ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**